libledger/DBInitializer.cpp

```diff
@@ -142,33 +141,27 @@ void DBInitializer::initLevelDBStorage()
 void DBInitializer::initTableFactory2(Storage::Ptr _backend)
 {
     auto cachedStorage = std::make_shared<CachedStorage>();
-#if 0
-    if (g_BCOSConfig.diskEncryption.enable)
-    {
-        auto encryptedStorage = std::make_shared<EncryptedStorage>();
-        encryptedStorage->setBackend(_backend);
-
-        encryptedStorage->setDataKey(asBytes(g_BCOSConfig.diskEncryption.dataKey));
-        cachedStorage->setBackend(encryptedStorage);
-    }
-    else
-    {
-#endif
     cachedStorage->setBackend(_backend);
-#if 0
-    }
-#endif
 
     cachedStorage->setMaxCapacity(
         m_param->mutableStorageParam().maxCapacity * 1024 * 1024);  // Bytes
     cachedStorage->setMaxForwardBlock(m_param->mutableStorageParam().maxForwardBlock);
 
     cachedStorage->init();
+    auto binaryLogStorage = make_shared<BinaryLogStorage>();
+    binaryLogStorage->setBackend(cachedStorage);
 
     auto tableFactoryFactory = std::make_shared<dev::storage::MemoryTableFactoryFactory2>();
-    tableFactoryFactory->setStorage(cachedStorage);
+    if (m_param->mutableStorageParam().binaryLog)

```

Comment:
The setting of binaryLog affects disableWAL in rocksDB commit.
Please add this logic.